web/viewer.js

```diff
@@ -1061,9 +1062,13 @@ var PDFViewerApplication = {
     });
   },
 
-  setInitialView: function pdfViewSetInitialView(storedHash, scale) {
+  setInitialView: function pdfViewSetInitialView(storedHash, scale, rotation) {
     this.isInitialViewSet = true;
 
+    if (rotation && rotation !== this.pageRotation) {
+      this.rotatePages(rotation);
```

Comment:
Also, comment https://github.com/mozilla/pdf.js/pull/5949/files#r49405341 is probably relevant here.
(Since the `this.rotatePages` call might cause us to trigger, and then cancel, unnecessary rendering operations on load.)